packages/next/build/entries.ts

```diff
@@ -421,6 +421,17 @@ export async function createEntrypoints(params: CreateEntrypointsParams) {
         return require.resolve(absolutePagePath)
       })()
 
+      /**
+       * When we find a middleware that is declared in the pages/ root we fail.
+       * There is no need to check on `dev` as this should only happen when
+       * building for production.
+       */
+      if (/[\\\\/]_middleware$/.test(page) && page !== '/_middleware') {
+        throw new Error(
+          `nested Middleware is deprecated (found pages${page}) - https://nextjs.org/docs/messages/nested-middleware`
```

Comment:
```suggestion
          `nested Middleware is not allowed (found pages${page}) - https://nextjs.org/docs/messages/nested-middleware`
```
Not sure we should call this deprecated as that usually implies it's still allowed but it is a hard error. 

We should also probably collect all nested middleware before showing this error so the user doesn't keep hitting the error removing one by one.